### DISCUSSION TITLE
Should an issue be a soft requirement before opening a PR?

### DISCUSSION THREAD

**Discussion by mikofski** [pvlib/pvlib-python#1452] [Polls]
To focus our work and avoid false starts I wonder if we should add the virtual constraint of requiring an issue to be opened for discussion prior to opening PRs. Since it's a soft requirement, some PR's could be opened without an issue, but this would be vetted by the maintainers and/or experienced contributors, and new contributors could be encouraged to start with the issue before embarking on new work. This might also minimize the number of open PRs that need review, because an issue that doesn't get a lot of attention may not be a substantial priority for the community at large. On the other hand this could create a large amount of unresolved issues, so it could be a trade off.

**Comment by kandersolar** [comment]
Devil's advocate: #1414 is an example of a PR that I think we'd all agree didn't need an issue, but encouraging the submitter to file one anyway may have presented enough of a barrier that they'd drop the contribution entirely.  But that type of PR is probably less common than PRs that should have been prevented by discussion. 

@mikofski what would this soft requirement look like in practice?  Some text in the PR template?

**Comment by wholmgren** [comment]
We have guidance on this topic here: https://pvlib-python.readthedocs.io/en/stable/contributing.html#when-should-i-submit-a-pull-request

I don't see a need for updating it. I don't think there is any real chance that a soft requirement would actually focus work. In any case, I don't want to strictly adhere to it for my own PRs. If a PR comes along that we don't want and for which there is no issue, then we should feel free to politely close it.
